test/ace.spec.js

```diff
@@ -0,0 +1,145 @@
+const path = require('path')
+const { strictEqual: equal } = require('assert')
+const { spawn } = require('child_process')
+const { promisify } = require('util')
+const fs = require('fs')
+
+const stat = promisify(fs.stat)
+const access = promisify(fs.access)
+const writeFile = promisify(fs.writeFile)
+const rimraf = promisify(require('rimraf'))
+const mkdirp = promisify(require('mkdirp'))
+
+const isFile = path =>
+  new Promise(resolve => {
+    fs.stat(path, (err, stat) => {
+      resolve(err === null && stat.isFile())
+    })
+  })
+
+const isDirectory = path =>
+  new Promise(resolve => {
+    fs.stat(path, (err, stat) => {
+      resolve(err === null && stat.isDirectory())
+    })
+  })
+
+const generateProject = opts => {
+  return Object.assign(
+    {
+      'package.json': JSON.stringify(
+        {
+          name: 'example-project',
+          license: 'MIT',
+          main: 'src/main/webapp/index.js',
+          'context-path': '/my-app',
+        },
+        null,
+        2
+      ),
+      'pom.xml': '',
+      'src/main/webapp/index.js': `module.exports = () => 'hello, world'`,
+      'src/main/webapp/example.spec.js': `it('should pass', () => {})`,
```

Comment:
❓ Is there a way to get tooling on JavaScript like this? For instance, providing raw JavaScript to a function that will simply stringify it?